crates/lib3h/src/dht/mirror_dht.rs

```diff
@@ -29,32 +31,44 @@ pub struct MirrorDht {
     inbox: VecDeque<DhtCommand>,
     /// Storage of EntryData with empty aspect content?
     entry_list: HashMap<Address, HashSet<Address>>,
-    /// Storage of PeerData
-    peer_list: HashMap<String, PeerData>,
+    /// Monotonic Storage of PeerData
+    peer_map: HashMap<String, PeerData>,
+    /// Track if peer timed out
+    timed_out_map: HashMap<String, HasTimedOut>,
```

Comment:
You could use a `HashSet<String>` instead for simplicity here, but if you intend to store more info in the future this pattern might be preferred.